src/ejabberd_c2s.erl

```diff
@@ -435,8 +435,29 @@ filter_mechanism(<<"EXTERNAL">>, S) ->
         error -> false;
         _ -> true
     end;
+filter_mechanism(<<"SCRAM-SHA-1-PLUS">>, S) ->
```

Comment:
There was some internal struggle with this for me as well :D but as filter_mechanism was already there I just extended it slightly. The problem with c2s is that it knows too much. I was looking for alternatives but it would be either passing lots of arguments or whole c2s state. The second option is not pretty as would require going back to c2s to extract values from the state. Either way moving these blocks of code would require bigger refactoring for which I don't feel competent enough with c2s to attempt.